rfcs/0013-ai-policies.md

```diff
@@ -0,0 +1,486 @@
+  HR->>WL: Forward request
+
+  %% JWT authentication check via Authorino
+  WL->>A: Validate JWT token & group access
+  alt Auth success
+    A-->>WL: Auth valid
+  else Auth fail
+    A-->>WL: Auth failed
+    WL->>HR: Return authentication error response (Error Response)
+    note right of WL: processing stops here for auth error
+  end
+
+  %% pre-model-server token rate limiting check
+  WL->>L: Check if limits reached
+  alt Limit not reached
+    L-->>WL: Rate limit OK
+  else Limit reached
+    L-->>WL: Rate limit exceeded
+    WL->>HR: Return rate limit exceeded response (Error Response)
+    note right of WL: processing stops here for rate limit error
+  end
+
+  %% forward request for inference if no early errors above
+  WL->>MS: Forward request for inference
+  MS-->>WL: Return model response with usage metrics
+
+  %% process response: parse usage metrics from model response
+  WL->>RPM: Parse usage metrics
+
+  %% update token usage count via Limitador
+  RPM->>L: Increment token usage count via limitador
+  L-->>RPM: Acknowledge token count update
+
+  %% final inference response: deliver back to client
+  RPM-->>WL: Return modified response
+  WL->>HR: Pass response for flush
+  HR->>C: Deliver final LLM response
+
+```
+
+
+#### Sequence Diagram: Prompt and Completion Response Guarding
+
+```mermaid
+sequenceDiagram
+  participant C as Client
+  participant HR as HTTP Route
+  participant WL as WASM/EnvoyFilter in Envoy (Guard Processor)
+  participant PG as LLM Prompt Guardian
+  participant MS as KServe Model Server
+  participant RG as LLM Response Guardian
+
+  %% initial req
+  C->>HR: Incoming chat/completion request
+  HR->>WL: Forward request
+
+  %% prompt risk check
+  WL->>PG: Initiate prompt risk check
+  alt Prompt Risk Check Passed
+    PG-->>WL: Risk check OK
+  else Prompt risk failed
+    PG-->>WL: Risk check failed
+    WL->>HR: Return prompt guard error response (Error Response)
+    note right of WL: Processing stops here for prompt risk error
+  end
+
+  %% forward request for inference if no early errors above
+  WL->>MS: Forward request for inference
+  MS-->>WL: Return model response
+
+  %% response risk check
+  WL->>RG: Initiate response risk check
+  alt Response Risk Check Passed
+    RG-->>WL: Risk check OK
+  else Response risk check failed
+    RG-->>WL: Risk check failed
+    WL->>HR: Return LLM response risk error response (Error Response)
+    note right of WL: Processing stops here for response risk error
+  end
+
+  %% final inference response: deliver back to client
+  WL->>HR: Forward final response
+  HR->>C: Deliver final LLM response
+```
+
+### Parsing OpenAI-style usage metrics
+
+OpenAI-style usage metrics for both completion and response APIs generally have a `usage` object, with values for `prompt_tokens` (token count for the initial prompt), `completion_tokens` (tokens generated by the model in response) and `total_tokens` (prompt + response tokens count).
```

Comment:
is there one of these that is more popular than the other? I am wondering does it make sense to support just one for now and expand beyond that in the future?

Reply:
assume this is re: Chat completion API vs Responses API

The chat completion API is more universally supported, the Responses API is newer, but is designed to work with more use-cases (agentic use-cases, as well as support for "reasoning/show my thoughts" response streaming)